plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/internal/commands/draw/DrawCoordinateCommand.java

```diff
@@ -0,0 +1,244 @@
+    /**
+     * @return Returns the line style.
+     */
+    public int getLineStyle() {
+        return style;
+    }
+
+    /**
+     * @return Returns the line width.
+     */
+    public int getLineWidth() {
+        return width;
+    }
+
+    /**
+     * Sets the line style
+     * 
+     * @param lineStyle the style of the line
+     * @param lineWidth the width of the line
+     */
+    public void setStroke(int lineStyle, int lineWidth) {
+        this.style = lineStyle;
+        this.width = lineWidth;
+    }
+
+    /**
+     * Sets the color that the shape will be filled with. If fill is null then no fill will be
+     * applied.
+     * 
+     * @param fillColor a color to be used to fill the shapeor null.
+     */
+    public void setFill(Color fillColor) {
+        this.fill = fillColor;
+    }
+
+
+    /**
+     * @return the useCircle
+     */
+    public boolean isUseCircle() {
+        return useCircle;
+    }
+
+    /**
+     * @param useCircle the useCircle to set
+     */
+    public void setUseCircle(boolean useCircle) {
+        this.useCircle = useCircle;
+    }
+
+    @Override
+    public Rectangle getValidArea() {
+        if (coord != null) {
+            try {
+                Envelope bounds = new ReferencedEnvelope(coord.x - 3, coord.y - 3, coord.x + 3,
+                        coord.y + 3, crs).transform(getMap().getViewportModel().getCRS(), true);
+                double[] points = new double[] { bounds.getMinX(), bounds.getMinY(),
+                        bounds.getMaxX(), bounds.getMaxY() };
+                getMap().getViewportModel().worldToScreenTransform().transform(points, 0, points, 0,
+                        2);
+                return new Rectangle((int) points[0], (int) points[1],
+                        (int) Math.abs(points[2] - points[0]),
+                        (int) Math.abs(points[3] - points[1]));
+            } catch (TransformException e) {
+                if (!errorReported) {
+                    errorReported = true;
+                    ProjectUIPlugin.log(
+                            "error calculating valid area, this will not be reported again", e);
+                }
+                return null;
+            } catch (MismatchedDimensionException e) {
+                if (!errorReported) {
+                    errorReported = true;
+                    ProjectUIPlugin.log(
+                            "error calculating valid area, this will not be reported again", e);
+                }
+                return null;
+            } catch (FactoryException e) {
+                if (!errorReported) {
+                    errorReported = true;
+                    ProjectUIPlugin.log(
+                            "error calculating valid area, this will not be reported again", e);
+                }
+                return null;
+            }
+        }
+        return null;
+
+    }
+
+    public void setValid(boolean valid) {
+        super.setValid(valid);
+        if (!valid)
```

Comment:
Is it necessary to overwride setValid() here? My read of ViewportPainter is that commands get disposed from within `clearCommands`. I guess you can remove method setValid from class because no additional behavior is required here to implement (for example removing listeners).

Reply:
Indeed it is not. I will remove it in a following commit